### PR TITLE
test: add the create and delete plugin in drawer

### LINKED ISSUE #1411: test: delete button in plugin drawer
https://github.com/apache/apisix-dashboard/pull/1402
added a delete button to the drawer of the plugin in this PR, 
now we need to add a test for it. 

Related links on completing test development: 
https://github.com/apache/apisix-dashboard/blob/a34317193cece2532ed333aaaacfd3a6af7328c4/docs/front-end-e2e.md

Very thanks!

### PR DESCRIPTION
**Pull request by stu01509**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
Fix #1411

### New feature or improvement
- Describe the details and related test reports.

Add the cypress test case to testing the plugin delete in the drawer.
image

___
### Please add the corresponding test cases if necessary.

### CODE REVIEW

**Comment by juzhiyuan** [comment]
also cc @guoqqqi to review

File: web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  it('should visit plugin market', function () {
+    cy.visit('/');
+    cy.contains('Plugin').click();
+    cy.contains('Create').click();
+    cy.fixture('plugin-dataset.json').as('cases');
+    cy.get('@cases').then((cases) => {
+      cy.configurePlugins(cases);
+    });
+  });
```
Comment (guoqqqi) [inline web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js:37]: In this test we don't need to create all the plugins, it takes up more time than necessary, we just need to create a few plugin examples to test that the deletion functions properly.

File: web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js [hunk 2]
```diff
@@ -45,7 +45,6 @@ context('Create and Delete Plugin List', () => {
       .then(() => {
         cy.get('.ant-drawer-footer').contains('button', 'Delete').click();
         cy.contains('button', 'Confirm').click({ force: true });
-        cy.get(this.domSelector.empty).should('be.visible');
       });
```
Comment (guoqqqi) [inline web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js:48]: This assertion ensures that the plugin can be removed in its entirety properly, and I think it's best that we don't remove it

**Comment by guoqqqi** [comment]
Hi, We need to test the functionality not only on the plug-in page but also on the route page and the service page to make sure it works.

**Comment by LiteSun** [comment]
ping~

**Comment by nic-chen** [comment]
ping @stu01509

File: web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js (hunk 2 above)
Comment (juzhiyuan) [reply web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js:48]: any update here?

**Comment by stu01509** [comment]
> Hi, We need to test the functionality not only on the plug-in page but also on the route page and the service page to make sure it works.

Hi 

Actually, I only found it the plugin page, where could I find in the route page and services page?

**Comment by guoqqqi** [comment]
Hi,you can find it in the Create, Configure Plugin step.

**Comment by guoqqqi** [comment]
CI error: https://github.com/apache/apisix-dashboard/pull/1597/checks?check_run_id=2148899222
This CI error has been fixed in #1584.

**Comment by stu01509** [comment]
Hi @guoqqqi and @juzhiyuan 

Update the commit.

**Comment by nic-chen** [comment]
@stu01509  FE e2e test failed

**Comment by guoqqqi** [comment]
The CI error is because the plugin has not been removed. Please refer to：
`web/cypress/integration/plugin/create-edit-delete-plugin.spec.js`
image

**Comment by stu01509** [comment]
I updated the commit, but it seems like occurred the error in another test case.

File: web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable no-undef */
+
+context('Create and Delete Plugin List', () => {
```
Comment (guoqqqi) [inline web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js:19]:
```suggestion
context('Delete Plugin List with the Drawer', () => {
```

File: web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    });
+
+    cy.get('.ant-drawer-content').should('be.visible').within(() => {
+      cy.get('#disable').click();
+      cy.get('.ant-switch-checked').should('exist');
+    });
+
+    cy.contains('button', 'Submit').click();
+    cy.get('.ant-drawer-content', { timeout }).should('not.exist');
+  });
```
Comment (guoqqqi) [inline web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js:45]: Please use `this.domSelector...` and `this.data...`, refer to other test cases.

File: web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+  it('should delete the plugin in drawer', function () {
+    cy.visit('/plugin/list');
+    cy.get(this.domSelector.refresh).click();
+    cy.contains('button', 'Edit').click();
+    cy.get('.ant-drawer-footer').contains('button', 'Delete').click({ force: true });
```
Comment (guoqqqi) [inline web/cypress/integration/plugin/create-delete-in-drawer-plugin.spec.js:51]: The dom element can be moved to public data for other tests use.

File: web/cypress/integration/route/create-edit-delete-route.spec.js [hunk 6]
```diff
@@ -72,12 +73,25 @@ context('Create and Delete Route', () => {
...
+    cy.get('.ant-drawer-content', { timeout }).should('not.exist');
+
+    cy.contains('prometheus').parents('.ant-card-bordered').within(() => {
+      cy.get('button').click({ force: true });
     });
-    cy.contains('button', 'Cancel').click();
+
+    cy.get('.ant-drawer-footer').contains('button', 'Delete').click({ force: true });
+    cy.contains('button', 'Confirm').click({ force: true });
+
```
Comment (guoqqqi) [inline web/cypress/integration/route/create-edit-delete-route.spec.js:94]: Also use `this.domSelector..` and `this.data...`
I think another step should be added: check that the plugin was successfully removed after it was deleted.

File: web/cypress/integration/service/create-edit-delete-service.spec.js [hunk 7]
```diff
@@ -35,6 +37,26 @@ context('Create and Delete Service ', () => {
...
+    cy.contains('button', 'Submit').click();
+    cy.get('.ant-drawer-content', { timeout }).should('not.exist');
+
+    cy.contains('prometheus').parents('.ant-card-bordered').within(() => {
+      cy.get('button').click({ force: true });
+    });
+
+    cy.get('.ant-drawer-footer').contains('button', 'Delete').click({ force: true });
+    cy.contains('button', 'Confirm').click({ force: true });
+
```
Comment (guoqqqi) [inline web/cypress/integration/service/create-edit-delete-service.spec.js:59]: ditto

**Review by guoqqqi — CHANGES_REQUESTED** [review]

**Comment by stu01509** [comment]
Hi @guoqqqi 

Update the commit, please review it again.

**Comment by nic-chen** [comment]
@stu01509  please sync codes from branch master

**Comment by stu01509** [comment]
Hello @nic-chen 

Update the commit.

**Comment by LiteSun** [comment]
Hi, @guoqqqi, I remember that you developed this delete feature, please make sure that this test is fully covered

**Review by LiteSun — APPROVED** [review]

File: web/cypress/fixtures/selector.json [hunk 8]
```diff
@@ -11,14 +11,19 @@
   "errorNotification:": ".ant-notification-notice-error",
 
   "pluginCard": ".ant-card",
+  "pluginCardBordered": ".ant-card-bordered",
   "pageContent": ".ant-pro-page-container",
 
+  "drawerContent": ".ant-drawer-content",
```
Comment (guoqqqi) [inline web/cypress/fixtures/selector.json:17]:
```suggestion
  "drawer": ".ant-drawer-content",
```
We already have that variable, just leave one of the two.

**Comment by guoqqqi** [comment]
Hi, @stu01509 Again these pages need to be tested.
image

**Comment by guoqqqi** [comment]
As the function is in a public component, it can be tested once and there is no need to add tests for other pages. Thanks.

**Comment by LiteSun** [comment]
> As the function is in a public component, it can be tested once and there is no need to add tests for other pages. Thanks.

The tests corresponding to route and service are unnecessary? @guoqqqi

**Comment by stu01509** [comment]
Update the commit.

**Review by juzhiyuan — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Pull request merged**